test_client.py

```diff
@@ -0,0 +1,15 @@
+import echo_client
+
+
+def test_basic_entry():
+    assert echo_client.start_client("Hello") == "Received message: Hello"
+
+
+def test_long_entry():
+    msg = echo_client.start_client("This message is longer than buffer")
+    assert msg == "Received message: This message is longer than buffer"
+
+
+def test_unicode_entry():
+    msg = echo_client.start_client(unicode('Hello'))
```

Comment:
When you test unicode, test with special characters outside the ASCII plane so that you can verify that you've not missed any implicit encodings or decodings.  Testing with all ASCII characters proves nothing, sadly.